### ISSUE TITLE
Run e2e tests in parallel

### ISSUE THREAD

**Issue by aramase** [Azure/azure-workload-identity#144]
The proxy test and token exchange test currently use the same service account for tests and running in parallel will cause a race. I can add trust for a new service account and enable parallel again in a follow-up PR.

_Originally posted by @aramase in https://github.com/Azure/aad-pod-managed-identity/pull/142#discussion_r688029308_
